### DISCUSSION TITLE
Support for `tree` as Combobox's Popover

### DISCUSSION THREAD

**Discussion by jaknas** [ariakit/ariakit#3977] [Questions] [answered]
Hi 👋. 
I'm interested in creating a Combobox with `tree` inside the popup. Looks like this is supported by ARIA ([Combobox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/)) and [`ComboboxPopover`](https://ariakit.org/reference/combobox-popover) component itself, which assigns `role='treeitem'` to each of the popups children. I'd say it's a solid foundation for me to start looking into it, but I have a question if you foresee any obstacles here, could point to any existing examples or have recommendations regarding implementation. 

I'd probably be looking into re-using parts from the now stale [`Tree` components PR](https://github.com/ariakit/ariakit/pull/1018).

Thanks in advance!

**Comment by benrodrs** [comment, marked as answer]
We still plan to have a `Tree` component that will be compatible with `Combobox`. However, you can already build it using the [Composite](https://ariakit.org/components/composite) module, as demonstrated by the tree PR.

Maybe @diegohaz has a better idea, but I don't see any major obstacles here. Ensure that arrow key navigation doesn't conflict with caret navigation on the combobox input, as horizontal arrow keys are used to expand or collapse tree groups. This is already considered when building comboboxes with grids, so the solution might be similar.
